server/server.go

```diff
@@ -61,12 +62,40 @@ func requestHandler(router proxy.Router, logger *apexlog.Logger) func(http.Respo
 			}
 		}
 
+		var reader io.ReadCloser
+		var writer http.ResponseWriter
+		if reqres.AddCompressionType != util.CompressionTypeNone {
+			if reqres.Response.Header.Get(headerContentEncodingKey) == "gzip" {
+				reader, err = util.NewGzipDecodingReader(reqres.Response.Body)
+				if (err != nil) {
+					writeError(w, err)
+					return
+				}
+			} else {
+				reader = reqres.Response.Body
+			}
+
+			writer = NewEncodingResponseWriter(w, reqres.AddCompressionType)
+			header.Del(headerContentLengthKey)
+			header.Set(headerContentEncodingKey, util.ContentEncodingFromCompressionType(reqres.AddCompressionType))
+			vary := header.Get(headerVaryKey)
+			if len(vary) > 0 {
+				vary = vary + ", " + headerContentEncodingKey
```

Comment:
If the existing Vary header already includes Content-Encoding, will this add a second copy?

Reply:
Fixed.

Reply:
Sorry for missing this earlier and talking about Content-Encoding. The Vary header should include Accept-Encoding, not Content-Encoding.